lib/net.c

```diff
@@ -1085,6 +1085,63 @@ start_iface(struct gatekeeper_if *iface)
 	return ret;
 }
 
+static int
+init_net_stage1(void *arg)
```

Comment:
Shouldn't we allocate the memory pool on the fly? That is to say, when we encounter a new ``numa`` node, we allocate the corresponding memory pool. Since in the configuration file ``lua/gatekeeper_config.lua``, we will allocate ``lcores`` to different blocks on the fly. Initially, the ``numa_used[x]`` array should be all ``false``, the initialization below won't do anything.

Reply:
I don't think all of the elements should be `false`, as the order of events should be:

(1) `lua/gatekeeper_config.lua` invokes the network configuration, which calls `gatekeeper_init_network()` to allocate and initialize `numa_used` to be an array of all `false`.
(2) `lua/gatekeeper_config.lua` invokes the individual block configurations, which allocate lcores. Based on the lcores that are allocated, the `numa_used` array is set to true for the corresponding sockets for those lcores.
(3) Stage 1 begins, and the first function that is invoked in this stage is now `init_net_stage1()`, which allocates memory pools for only the NUMA nodes that were set to `true` in `numa_used` by the individual block configurations.

Reply:
You are right! It's launched at the first stage right after the Lua configuration finishes.